### DISCUSSION TITLE
Check of end of file on BitStream?

### DISCUSSION THREAD

**Discussion by daveisfera** [scott-griffiths/bitstring#348] [Q&A]
Is there a way to check for end of file on a `BitStream`? Right now, I'm doing `bs._pos >= len(bs)` but then it's accessing a private member and it feels like there should be an easier way to check for this.

**Comment by scott-griffiths** [comment]
Hi. There should be no need to access the private member as you can just use `bs.pos` instead. This will also prevent you from getting to a position past the end of the file so the test becomes `bs.pos == len(bs)`.

I think is a reasonable way to check if you're at the end of the `BitStream`.

**Reply by daveisfera** [reply]
Oh, I hadn't noticed that property. I was looking through the `def` statements and didn't notice those property definitions at the end. I'll start using that, but might be worth adding an `eof` function or something like that to simplify/abstract this check.

Thanks for your help!
